jade/js/search.js

```diff
@@ -1,126 +1,117 @@
-$(document).ready(function() {
-
-  $("#clear-search").click(function() {
-    searchMatchesFound = 0
-    $("input#search[type=text], textarea").val("");
-    $("#search-msg").remove();
-    $(".search-results").children().remove();
-    $("input#search").focus();
-  });
-
-  $("input#search").on("click", function() {
-    $("input#search").focus();
-
-    $.when($(".category-msg").fadeOut()).done(function() {
-        $("#search").attr("placeholder", "Search Applications...");
-        $(".category-container, #main-dashboard").hide();
-        $("#search-icon").addClass("animated slideInLeft");
-        $(".search-results, #search-icon").show();
-        $(".dash-btn").fadeIn();
-        grid(".search-results");
-      });
-  });
-
-  // Init a timeout variable to be used below
-  var timeout = null;
-
-  // Listen for key up events
-  $("#search").on("keyup", function() {
-
-    // fix return to search
-    if ($("#main-dashboard").css("display") == 'block') { // don't repeat animation
-            $("#main-dashboard").fadeOut("slow", function() {
-              $(".search-results, #search-icon, dash-btn").fadeIn("slow");
-            });
-                };
-
-    // Assign search value to input
-    searchValue = this.value;
-
-    // Clear the timeout if it has already been set.
-    // This will prevent the previous task from executing
-    // if it has been less than <MILLISECONDS>
-    clearTimeout(timeout);
-
-    // Make a new timeout set to go off in 1000ms
-    // This depends how fast people type, it should be ok even for slow typers
-    timeout = setTimeout(function() {
-
-      if (searchValue.length !== 0) {
-
-        $(".search-results").empty();
+$(document).ready(function () {
+    var search = $("#search");
```

Comment:
Preload some search results can speedup application.

Reply:
If i dont empty the div it will append search results on top of existing results, we probably can preload but we need to remove the ones that dont match.

Reply:
Yes, but u can empty search results without searching div every time.